api/v1alpha1/envoyfleet_types.go

```diff
@@ -126,6 +132,21 @@ type AccessLoggingConfig struct {
 	JsonTemplate map[string]string `json:"json_template,omitempty"`
 }
 
+type TLS struct {
+	CipherSuites              []string           `json:"cipherSuites,omitempty"`
+	TlsMinimumProtocolVersion string             `json:"tlsMinimumProtocolVersion,omitempty"`
+	TlsMaximumProtocolVersion string             `json:"tlsMaximumProtocolVersion,omitempty"`
+	TlsSecretsConfig          []TLSSecretsConfig `json:"tlsSecretsConfig"`
```

Comment:
it is ok to have `json:"tlsSecrets"` since this will be visible to the user. I.e. it reads as "tls certificates secrets", "tlsSecretsConfig" will be more confusing. In fact we can completely rename the type to TLSSecrets as well.